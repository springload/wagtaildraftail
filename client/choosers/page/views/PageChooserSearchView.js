import React from 'react';
import PropTypes from 'prop-types';

import PageChooserResultSet from '../PageChooserResultSet';

// TODO Get rid of any propTypes.
// TODO Figure out whether those values are required or not.
const propTypes = {
  totalItems: PropTypes.number.isRequired,
  pageNumber: PropTypes.any,
  totalPages: PropTypes.any,
  items: PropTypes.any,
  pageTypes: PropTypes.any,
  restrictPageTypes: PropTypes.any,
  onPageChosen: PropTypes.any,
  onNavigate: PropTypes.any,
  onChangePage: PropTypes.any,
};

const renderTitle = (totalItems) => {
  switch (totalItems) {
  case 0:
    return 'There are no matches';
  case 1:
    return 'There is 1 match';
  default:
    return `There are ${totalItems} matches`;
  }
};

class PageChooserSearchView extends React.Component {
  render() {
    const {
      totalItems,
      pageNumber,
      totalPages,
      items,
      pageTypes,
      restrictPageTypes,
      onPageChosen,
      onNavigate,
      onChangePage,
    } = this.props;
    return (
      <div className="nice-padding">
        <h2>{renderTitle(totalItems)}</h2>
        <PageChooserResultSet
          pageNumber={pageNumber}
          totalPages={totalPages}
          items={items}
          pageTypes={pageTypes}
          restrictPageTypes={restrictPageTypes}
          onPageChosen={onPageChosen}
          onNavigate={onNavigate}
          onChangePage={onChangePage}
        />
      </div>
    );
  }
}

PageChooserSearchView.propTypes = propTypes;
// PageChooserSearchView.defaultProps = defaultProps;

export default PageChooserSearchView;
