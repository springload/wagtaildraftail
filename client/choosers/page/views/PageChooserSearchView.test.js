import React from 'react';
import { shallow } from 'enzyme';
import PageChooserSearchView from './PageChooserSearchView';

describe('PageChooserSearchView', () => {
  it('renders', () => {
    expect(shallow((
      <PageChooserSearchView
        onPageChosen={() => {}}
        onNavigate={() => {}}
        pageTypes={{}}
        totalItems={0}
      />
    ))).toMatchSnapshot();
  });
});
