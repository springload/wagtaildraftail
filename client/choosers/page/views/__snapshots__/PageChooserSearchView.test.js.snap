// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageChooserSearchView renders 1`] = `
<div
  className="nice-padding"
>
  <h2>
    There are no matches
  </h2>
  <PageChooserResultSet
    displayChildNavigation={false}
    items={Array []}
    onNavigate={[Function]}
    onPageChosen={[Function]}
    pageTypes={Object {}}
    parentPage={null}
    restrictPageTypes={Array []}
  />
</div>
`;
